src/components/navbar-editor.tsx

```diff
@@ -471,56 +525,119 @@ export default function EditorNavbar(props: EditorNavbarProps) {
+			// this fixed it by updating the branch before making it so there is only the 2 file needed for PR
+			// I am commenting this out for now and tackle this later if needed / if there been this senario issue
+			try {
+				await synchronizeForkWithUpstream(accessToken, forkedRepo.owner.login, forkedRepo.name);
+			} catch (error) {
+				reportMetric("github_publish.failure.sync_with_upstream");
+				console.warn("Failed to sync fork with upstream: ", error);
+			}
+			*/
+
 			const latestCommitSha = await fetchLatestCommitSha(accessToken, forkedRepo.owner.login, forkedRepo.name, forkedRepo.default_branch);
 			if (!latestCommitSha) {
+				reportMetric("github_publish.failure.commit_sha");
 				throw new Error("Failed to fetch the latest commit SHA.");
 			}
 
 			const newBranchName = `Automated-PR-${Date.now()}`;
-
-			await createBranch(accessToken, forkedRepo.owner.login, forkedRepo.name, newBranchName, latestCommitSha);
+			try {
+				await createBranch(accessToken, forkedRepo.owner.login, forkedRepo.name, newBranchName, latestCommitSha);
+			} catch (error) {
+				reportMetric("github_publish.failure.branch");
+				throw new Error("Failed to create branch: " + (error instanceof Error ? error.message : String(error)));
+			}
 
 			const imageBase64 = thumbnailPreview.value || null;
-			const imageBlobSha = imageBase64 ? await createBlobForImage(accessToken, forkedRepo.owner.login, forkedRepo.name, imageBase64.split(',')[1]) : null;
-
-			const treeSha = await createTreeAndCommit(
-				accessToken,
-				forkedRepo.owner.login,
-				forkedRepo.name,
-				latestCommitSha,
-				[
-					{ path: `games/${gameTitle}.js`, content: gameCode },
-					...(imageBlobSha ? [{ path: `games/img/${gameTitle}.png`, sha: imageBlobSha }] : [])
-				]
-			);
-
-			const newCommit = await createCommit(accessToken, forkedRepo.owner.login, forkedRepo.name, `Automated Commit - ${gameTitle}`, treeSha, latestCommitSha);
-
-			await updateBranch(accessToken, forkedRepo.owner.login, forkedRepo.name, newBranchName, newCommit.sha);
-
-			const pr = await createPullRequest(
-				accessToken,
-				"hackclub",
-				"sprig",
-				`[Automated] ${gameTitle}`,
-				newBranchName,
-				"main",
-				`### Author name\nAuthor: ${authorName}\n\n### About your game\n\n**What is your game about?**\n${gameDescription}\n\n**How do you play your game?**\n${gameControlsDescription}`,
-				forkedRepo.owner.login,
-				gameID ?? ''
-			);
-
-			githubPRUrl.value = pr.html_url;
+			let imageBlobSha = null;
+			try {
+				if (imageBase64) {
+					imageBlobSha = await createBlobForImage(accessToken, forkedRepo.owner.login, forkedRepo.name, imageBase64.split(',')[1]);
+				}
+			} catch (error) {
+				reportMetric("github_publish.failure.image_blob");
+				throw new Error("Failed to create image blob: " + (error instanceof Error ? error.message : String(error)));
+			}
 
-			publishSuccess.value = true;
+			const sanitizedGameTitle = gameTitle.replace(/\s+/g, '-');
```

Comment:
Is this needed?

Reply:
We are just doing this for filename, not the game title restrictions